skyplane/replicate/replicator_client.py

```diff
@@ -427,6 +387,54 @@ def run_replication_plan(
                     chunks.append(chunk)
                     idx += 1
 
+            # initiate multipart transfers in parallel
+            progress.update(prepare_task, description=f": Queuing {len(multipart_pairs)} files for multipart transfers")
+            obj_store_interface = ObjectStoreInterface.create(job.dest_region, job.dest_bucket)
+            upload_ids = do_parallel(lambda x: obj_store_interface.initiate_multipart_upload(x[1].key), multipart_pairs, n=16)
+            for (src_object, dest_object), upload_id in upload_ids:
+                # determine number of chunks via the following algorithm:
+                chunk_size_bytes = int(multipart_min_size_mb * MB)
+                num_chunks = math.ceil(src_object.size / chunk_size_bytes)
+                if num_chunks > multipart_max_chunks:
+                    chunk_size_bytes = int(src_object.size / multipart_max_chunks)
+                    chunk_size_bytes = math.ceil(chunk_size_bytes / MB) * MB  # round to next largest mb
+                    num_chunks = math.ceil(src_object.size / chunk_size_bytes)
+                offset = 0
+                part_num = 1
+                parts = []
```

Comment:
could we just do `parts = range(1, num_chunks+1, 1)` and get rid of `parts_num`?

Reply:
We could, but there is an assert statement in that loop that checks if the file size is greater than 0 before adding to the parts list, so I'm not sure that it is best to make this change.

Reply:
ah ok, sounds good